tests/cpp_unit_tests/test_current_sensor.cpp

```diff
@@ -33,92 +33,142 @@ TEST_CASE("Test current sensor") {
+
+                // Check symmetric sensor output for asymmetric parameters
+                CHECK(asym_sensor_param.measurement.real_component.variance[0] ==
+                      doctest::Approx(0.5 * (i_variance_pu + i_angle_variance_pu * i_pu * i_pu)));
+                auto const shifted_i_angle = i_angle + deg_240;
+                CHECK(asym_sensor_param.measurement.imag_component.variance[1] ==
+                      doctest::Approx(i_variance_pu * sin(shifted_i_angle) * sin(shifted_i_angle) +
+                                      i_angle_variance_pu * i_pu * i_pu * cos(shifted_i_angle) * cos(shifted_i_angle)));
+                CHECK(real(asym_sensor_param.measurement.value()[0]) == doctest::Approx(i_pu * cos(i_angle)));
+                CHECK(imag(asym_sensor_param.measurement.value()[1]) == doctest::Approx(i_pu * sin(shifted_i_angle)));
+
+                CHECK(sym_sensor_output_asym_param.id == 0);
+                CHECK(sym_sensor_output_asym_param.energized == 1);
+                for (auto phase = 0; phase < 3; ++phase) {
+                    CAPTURE(phase);
+                    if (angle_measurement_type == AngleMeasurementType::global_angle) {
+                        CHECK(sym_sensor_output_asym_param.i_residual[phase] == doctest::Approx(0.0));
+                        CHECK(sym_sensor_output_asym_param.i_angle_residual[phase] == doctest::Approx(0.0));
+                    } else {
+                        CHECK(sym_sensor_output_asym_param_local_residuals.i_residual[phase] == doctest::Approx(0.0));
+                        CHECK(sym_sensor_output_asym_param_local_residuals.i_angle_residual[phase] ==
+                              doctest::Approx(0.0));
+                    }
+                }
+
+                CHECK(sym_current_sensor.get_terminal_type() == terminal_type);
+                if (angle_measurement_type == AngleMeasurementType::global_angle) {
+                    CHECK(sym_current_sensor.get_angle_measurement_type() == AngleMeasurementType::global_angle);
+                } else {
+                    CHECK(sym_current_sensor.get_angle_measurement_type() == AngleMeasurementType::local_angle);
+                }
+            }
         }
         SUBCASE("Wrong measured terminal type") {
             for (auto const terminal_type :
                  {MeasuredTerminalType::source, MeasuredTerminalType::shunt, MeasuredTerminalType::load,
                   MeasuredTerminalType::generator, MeasuredTerminalType::node}) {
-                CHECK_THROWS_AS((CurrentSensor<symmetric_t>{
-                                    {1, 1, terminal_type, AngleMeasurementType::local_angle, 1.0, 1.0, 1.0, 1.0}, 1.0}),
-                                InvalidMeasuredTerminalType);
+                CHECK_THROWS_AS(
+                    (CurrentSensor<symmetric_t>{
+                        {1, 1, terminal_type, AngleMeasurementType::global_angle, 1.0, 1.0, 1.0, 1.0}, 1.0}),
+                    InvalidMeasuredTerminalType);
```

Comment:
why did you need this change? if you do, why not make it
```suggestion
                CHECK_THROWS_AS(
                    (CurrentSensor<symmetric_t>{
                        {1, 1, terminal_type, measured_angle_type, 1.0, 1.0, 1.0, 1.0}, 1.0}),
                    InvalidMeasuredTerminalType);
```

Reply:
I just realized I forgot to modify other parts of the test... This was for when only global angle was implemented. Good catch.